BREAKING.md

```diff
@@ -57,6 +60,39 @@ getFileLink(
 )
 ```
 
+### AgentScheduler removed as a default member of ContainerRuntime
+Prior to 0.38, `ContainerRuntime` would automatically add an `AgentScheduler` to the container upon creation.  This is no longer added by default.  If you support back compat with documents created prior to 0.38, they will require the `AgentScheduler` to be present in the registry for compatibility.
+
+To facilitate migration, `makeContainerRuntimeWithAgentScheduler()` is provided in the `@fluidframework/deprecated-agent-scheduler-container-runtime` package that produces a runtime which is compatible with the old `ContainerRuntime.load()` by adding an `AgentScheduler`.  This helper is deprecated for new use and will be removed in an upcoming release.  Container authors should plan to incorporate the `AgentScheduler` into their container code if needed, and move to the new `ContainerRuntime.load()`.
+
+`AgentSchedulerBaseContainerRuntimeFactory` and `AgentSchedulerContainerRuntimeFactoryWithDefaultDataStore` are also provided for back-compat with the aqueduct container runtime factories.  Similarly to `makeContainerRuntimeWithAgentScheduler()`, container authors should plan to incorporate the `AgentScheduler` into their subclasses (adding it to the registry and instantiating on `containerInitializingFirstTime()`).
```

Comment:
I assume Bohemia needs these. If so, it would be great to spell out next steps somewhere. I.e. I'd hope that "deprecated" actually means we are deleting it in couple releases, and the guidance to consumers - if you need it - copy code, it will be gone. Otherwise we will run into same problem as in another places API marked deprecated for years with no way to delete them (cause there is usage in Bohemia)

Reply:
I don't know of any core bohemia code that uses this. I did a quick search for _scheduler in our repo and I see the task path might use it for something. Whoever is pushing forth this change should work with them to remove their dependency on the scheduler as it was always a bad/not recommended pattern to have a dependency on a hardcoded id singleton (what if the container didn't bring it), and the office-fluid-container itself likely has no reason to bring in the scheduler, especially with this pattern. If we do want to bring in something like this it should be a container service or data producer/consumer, not some random singleton.

I would prefer to not have to do that hack in the container for component specific requirements though and would instead greatly prefer whoever is working with the tasks team to recommend against such patterns and design better ones :)

Reply:
If nothing else, the planner components themselves can just initialize their own version of the AgentScheduler object to use the functionality they want.